[AB#37473] Fix pagination docs (#477)
Amsterdam/dso-api: docs/source/generic/rest/pagination.rst

Met de velden ``_links.next`` en ``_links.previous`` zijn respectievelijk de volgende en vorige pagina op te vragen.
Meer algemeen kan pagina `n` worden opgevraagd met :samp:`?page={n}`.
Paginanummers beginnen bij één, niet nul.

> De links ``next`` en ``previous`` ontbreken op de eerste, resp. laatste, pagina.

Author: ```suggestion
De links ``next`` en ``previous`` ontbreken op de laatste, resp. eerste, pagina.
```

In het object ``page`` zijn de volgende velden opgenomen:

> * ``page.number``: het huidige paginanummer.

Author: ```suggestion
* ``page.number``: het huidige paginanummer;
```

* ``page.size``: de grootte van een pagina.

> Wordt ``_count=true`` meegegeven, dat bevat de response tevens

Reviewer: ```suggestion
Wordt ``?_count=true`` meegegeven, dat bevat de response tevens:
```

Die `?` notatie wordt (bijna) overal gebruikt om te duiden dat het een query-string parameter is.

> Wordt ``?_count=true`` meegegeven, dat bevat de response tevens:

Author: ```suggestion
Wordt ``?_count=true`` meegegeven, dat bevat het tevens:
```

> * ``page.totalElements``: het aantal objecten in de (gefilterde) resultaatset,

Author: ```suggestion
* ``page.totalElements``: het aantal objecten in de (gefilterde) resultaatset;
```

* ``page.totalPages``: het aantal pagina's dat de resultaatset beslaat.

Bijvoorbeeld:

.. code-block:: javascript
    "page": {
        "number": 1,
        "size": 20,
        "totalElements": 117,
        "totalPages": 6,
    }

> Het tellen van resultaten moet expliciet worden aangevraagd omdat het een potentieel
> dure operatie is, vooral als er geen filters worden gebruikt.

Reviewer: ```suggestion
Het tellen van resultaten moet expliciet worden aangevraagd omdat het bij grote datasets
erg vertragend werkt, terwijl deze gegevens zelden gebruikt worden.
```

"dure" is iets te technisch.

Author: "terwijl deze gegevens zelden gebruikt worden" laat ik liever weg. Als de gebruiker die gegeven wil hebben is dat niet relevant, alleen of de boel daardoor trager wordt.

Reviewer: prima! Ik heb de comment nog iets geedit naar "zelden nodig zijn". Dat is inderdaad nogal een aanname, weliswaar wel op basis van het huidige gebruik.

Reviewer: Ook bleek uit de blogpost dat `_links.next` uitlezen een denkstap is die menigeen niet gemaakt heeft. Dan wordt er eerder gekeken "if resultaten < totaal" om vervolgens `_links.next` uit te lezen.

Author: Ok, dan commit ik dit met een commit eroverheen.

> De velden uit het ``page`` object worden ook als HTTP-headers in de response teruggegeven:

Author: ```suggestion
De velden uit het ``page``-object worden ook als HTTP-headers in de response teruggegeven:
```

Met de velden ``_links.next`` en ``_links.previous`` zijn respectievelijk de volgende en vorige pagina op te vragen.
Meer algemeen kan pagina `n` worden opgevraagd met :samp:`?page={n}`.
Paginanummers beginnen bij één, niet nul.
De links ``next`` en ``previous`` ontbreken op de laatste, resp. eerste, pagina.

In het object ``page`` zijn de volgende velden opgenomen:

* ``page.number``: het huidige paginanummer;
* ``page.size``: de grootte van een pagina.

Wordt ``?_count=true`` meegegeven, dat bevat het tevens:

* ``page.totalElements``: het aantal objecten in de (gefilterde) resultaatset;
* ``page.totalPages``: het aantal pagina's dat de resultaatset beslaat.

Bijvoorbeeld:

.. code-block:: javascript
    "page": {
        "number": 1,
        "size": 20,
        "totalElements": 117,
        "totalPages": 6,
    }

Het tellen van resultaten moet expliciet aangevraagd worden omdat dit bij grote

> datasets erg vertragend werkt, terwijl die gegevens zelden nodig zijn.
> Om te weten of er een volgende pagina is, kan je beter de `_links.next` uitlezen.

Author: ```suggestion
datasets vertragend werkt. Lees `_links.next` uit om te zien of er een volgende pagina is.
```
